Build fail on Docusaurus 3.5.1

Hello, first, thank you for this plugin !

I just upgraded to the newest version of Docusaurus (3.5.1) but I can't build with it :

```bash
[ERROR] Error: Unable to build website for locale fr.
    at tryToBuildLocale (/node_modules/.pnpm/@docusaurus+core@3.5.1_@docusaurus+types@3.5.1_react-dom@18.3.1_react@18.3.1__react@18.3.1__d_upty6jj6xlpwwmkc3vds73poq4/node_modules/@docusaurus/core/lib/commands/build.js:54:19)
    at async /node_modules/.pnpm/@docusaurus+core@3.5.1_@docusaurus+types@3.5.1_react-dom@18.3.1_react@18.3.1__react@18.3.1__d_upty6jj6xlpwwmkc3vds73poq4/node_modules/@docusaurus/core/lib/commands/build.js:65:9
    at async mapAsyncSequential (/node_modules/.pnpm/@docusaurus+utils@3.5.1_@docusaurus+types@3.5.1_react-dom@18.3.1_react@18.3.1__react@18.3.1__typescript@5.4.5/node_modules/@docusaurus/utils/lib/jsUtils.js:21:24)
    at async Command.build (/node_modules/.pnpm/@docusaurus+core@3.5.1_@docusaurus+types@3.5.1_react-dom@18.3.1_react@18.3.1__react@18.3.1__d_upty6jj6xlpwwmkc3vds73poq4/node_modules/@docusaurus/core/lib/commands/build.js:63:5) {
  [cause]: TypeError: Cannot read properties of undefined (reading 'load')
      at parse (/node_modules/.pnpm/@easyops-cn+docusaurus-search-local@0.39.0_@docusaurus+theme-common@3.5.1_@docusaurus+plugin-_btcdilkjrraiu2f6hyizxb3upy/node_modules/@easyops-cn/docusaurus-search-local/dist/server/server/utils/parse.js:10:33)
      at /node_modules/.pnpm/@easyops-cn+docusaurus-search-local@0.39.0_@docusaurus+theme-common@3.5.1_@docusaurus+plugin-_btcdilkjrraiu2f6hyizxb3upy/node_modules/@easyops-cn/docusaurus-search-local/dist/server/server/utils/scanDocuments.js:24:46
      at Generator.next (<anonymous>)
      at fulfilled (/node_modules/.pnpm/tslib@2.6.3/node_modules/tslib/tslib.js:166:62)
}
```

Docusaurus version: 3.5.1
Node version: v20.11.1

It looks like it can't `load` something with the cheerio package, but I have no idea what it is.

I tried to change the locale (both in Docusaurus and plugin configuration) but got the same error with `en` locale.

--- a/.eslintrc
+++ b/.eslintrc
@@ -24,7 +24,7 @@
   },
   "overrides": [
     {
-      "files": ["*.config.js", "__jest__/**/*.js"],
+      "files": ["jestSetup.js", "*.config.js", "__jest__/**/*.js"],
       "env": {
         "node": true,
         "browser": false

--- a/docusaurus-search-local/package.json
+++ b/docusaurus-search-local/package.json
@@ -37,7 +37,7 @@
     "@docusaurus/utils-validation": "^2 || ^3",
     "@easyops-cn/autocomplete.js": "^0.38.1",
     "@node-rs/jieba": "^1.6.0",
-    "cheerio": "^1.0.0-rc.3",
+    "cheerio": "^1.0.0",
     "clsx": "^1.1.1",
     "debug": "^4.2.0",
     "fs-extra": "^10.0.0",

--- a/docusaurus-search-local/src/server/utils/getCondensedText.spec.ts
+++ b/docusaurus-search-local/src/server/utils/getCondensedText.spec.ts
@@ -1,4 +1,4 @@
-import cheerio from "cheerio";
+import * as cheerio from "cheerio";
 import { getCondensedText } from "./getCondensedText";
 
 describe("getCondensedText", () => {

--- a/docusaurus-search-local/src/server/utils/parse.ts
+++ b/docusaurus-search-local/src/server/utils/parse.ts
@@ -1,4 +1,4 @@
-import cheerio from "cheerio";
+import * as cheerio from "cheerio";
 import {
   ParsedDocument,
   ProcessedPluginOptions,

--- a/docusaurus-search-local/src/server/utils/parseDocument.spec.ts
+++ b/docusaurus-search-local/src/server/utils/parseDocument.spec.ts
@@ -1,4 +1,4 @@
-import cheerio from "cheerio";
+import * as cheerio from "cheerio";
 import { ParsedDocument } from "../../shared/interfaces";
 import { parseDocument } from "./parseDocument";
 

--- a/docusaurus-search-local/src/server/utils/parsePage.spec.ts
+++ b/docusaurus-search-local/src/server/utils/parsePage.spec.ts
@@ -1,4 +1,4 @@
-import cheerio from "cheerio";
+import * as cheerio from "cheerio";
 import { ParsedDocument } from "../../shared/interfaces";
 import { parsePage } from "./parsePage";
 

--- a/jest.config.js
+++ b/jest.config.js
@@ -12,6 +12,7 @@ module.exports = {
   moduleNameMapper: {
     "\\.module\\.css$": require.resolve("identity-obj-proxy"),
     "@docusaurus/Translate": "<rootDir>/__mocks__/Translate.js",
+    "^cheerio$": require.resolve("cheerio"),
   },
   // Ref https://github.com/facebook/jest/issues/2070#issuecomment-431706685
   // Todo(steve): remove next line when issue fixed.

--- a/jestSetup.js
+++ b/jestSetup.js
@@ -1,4 +1,8 @@
 import { matcherHint, printExpected, printReceived } from "jest-matcher-utils";
+import { TextEncoder, TextDecoder } from "node:util";
+import { ReadableStream } from "node:stream/web";
+
+Object.assign(global, { TextDecoder, TextEncoder, ReadableStream });
 
 const passMessage = (actual, expected) => () =>
   `${matcherHint(".not.toMatchPath")}

--- a/package.json
+++ b/package.json
@@ -25,6 +25,7 @@
     "@typescript-eslint/eslint-plugin": "^6.10.0",
     "@typescript-eslint/parser": "^6.10.0",
     "babel-jest": "^28.1.0",
+    "cheerio": "^1.0.0",
     "cross-env": "^7.0.3",
     "eslint": "^8.15.0",
     "eslint-config-prettier": "^8.5.0",

